src/d3a/models/market/one_sided.py

```diff
@@ -106,25 +110,34 @@ def accept_offer(self, offer_or_id: Union[str, Offer], buyer: str, *, energy: in
             if time is None:
                 time = self._now
             if energy is not None:
-                # Partial trade
                 if energy == 0:
                     raise InvalidTrade("Energy can not be zero.")
+                # partial energy is requested
                 elif energy < offer.energy:
                     original_offer = offer
+                    # Don't know why is it here
```

Comment:
It is used in order to select the correct hash value for the case of smart contract. If we are not running on blockchain, the offer.id is selected. If we are running on blockchain, the offer.real_id is selected, which is the hash id of the offer on the smart contract. If this is clear you can remove the comment.